tritium-core/src/test/java/com/palantir/tritium/event/InstrumentationPropertiesTest.java

```diff
@@ -82,4 +105,80 @@ public void invalid() {
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessage("name cannot be null or empty: {name=}");
     }
+
+    @Test
+    public void racingSystemProperties() throws Exception {
+        CountDownLatch latch = new CountDownLatch(8);
+        List<Callable<Object>> tasks = ImmutableList.of(
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "getSystemPropertySupplier: " + InstrumentationProperties.getSystemPropertySupplier("test")
+                            .asBoolean();
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "getProperty: " + System.getProperty("instrument.test");
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "setProperty: " + System.setProperty("instrument.test", "true");
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    for (int i = 0; i < 1000; i++) {
+                        System.setProperty("test" + i, "value" + i);
+                    }
+                    return "setProperties: " + System.getProperties();
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    InstrumentationProperties.reload();
+                    return "reload";
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "getSystemPropertySupplier: "
+                            + InstrumentationProperties.getSystemPropertySupplier("test").asBoolean();
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "isSpecificEnabled: " + InstrumentationProperties.isSpecificEnabled("test");
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "isGloballyEnabled: " + InstrumentationProperties.isGloballyEnabled();
+                });
+
+        List<ListenableFuture<Object>> futures = tasks.stream()
```

Comment:
nit: `executor.invokeAll(tasks)`

Reply:
[`ListeningExecutorService#invokeAll` unfortunately returns a `List<Future<T>>`](https://google.github.io/guava/releases/23.6-jre/api/docs/com/google/common/util/concurrent/ListeningExecutorService.html#invokeAll-java.util.Collection-) so requires an explicit cast to raw `List`:

```
        @SuppressWarnings("unchecked") // guaranteed by ListenableExecutorService
        List<ListenableFuture<Object>> futures = (List) executorService.invokeAll(tasks);
```

I switched over to that and adjusted the checks a bit